cdap-app-fabric/src/main/java/io/cdap/cdap/app/store/Store.java

```diff
@@ -220,6 +220,17 @@ ProgramDescriptor loadProgram(ProgramId program) throws IOException, Application
    */
   Map<ProgramRunId, RunRecordMeta> getRuns(ProgramRunStatus status, long startTime,
                                            long endTime, int limit, Predicate<RunRecordMeta> filter);
+  /**
+   * Fetches run records killed between startTimeSeconds and endTimeSeconds.
+   *
+   * @param applicationId the application id to match against
+   * @param startTimeSeconds fetch run history that has status last changed after the startTime
+   * @param endTimeSeconds fetch run history that has status last changed before the endTime
+   * @param limit max number of entries to fetch for this history call
+   * @return a map from {@link ProgramRunId} to the corresponding {@link RunRecordMeta}.
+   */
+  Map<ProgramRunId, RunRecordMeta> getRunsKilled(ApplicationId applicationId,
```

Comment:
Why need a specific method? Can't use the 
```
Map<ProgramRunId, RunRecordMeta> getRuns(ProgramRunStatus status, long startTime,
                                           long endTime, int limit, Predicate<RunRecordMeta> filter);
```

?

Reply:
No, we need to condition on the stop time in the runrecordmeta, this method conditions on the start time